File: leetcode/80.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 0
        x = len(nums)

        for i in range(x - 1, 0, -1):
            if nums[i] == nums[i - 1]:
                count += 1
                if count >= 2:
                    nums.pop(i)
                    count += 1
                    x -= 1
            else:
                count = 0

        return len(nums)

File: leetcode/80.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count = 0;
        int x = nums.size();

        for (int i = x - 1; i > 0; --i) {
            if (nums[i] == nums[i - 1]) {
                count++;
                if (count >= 2) {
                    nums.erase(nums.begin() + i);
                    count++;
                    x--;
                }
            } else {
                count = 0;
            }
        }

        return nums.size();
    }
};
